waiting_room.go

```diff
@@ -0,0 +1,148 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// WaitingRoom describes a WaitingRoom object.
+type WaitingRoom struct {
+	ID                    string    `json:"id,omitempty"`
+	CreatedOn             time.Time `json:"created_on,omitempty"`
+	ModifiedOn            time.Time `json:"modified_on,omitempty"`
+	Name                  string    `json:"name"`
+	Description           string    `json:"description,omitempty"`
+	Suspended             bool      `json:"suspended"`
+	Host                  string    `json:"host"`
+	Path                  string    `json:"path"`
+	QueueAll              bool      `json:"queue_all"`
+	NewUsersPerMinute     int       `json:"new_users_per_minute"`
+	TotalActiveUsers      int       `json:"total_active_users"`
+	SessionDuration       int       `json:"session_duration"`
+	DisableSessionRenewal bool      `json:"disable_session_renewal"`
+	CustomPageHtml        string    `json:"custom_page_html,omitempty"`
+}
+
+// WaitingRoomDetailResponse is the API response, containing a single WaitingRoom.
+type WaitingRoomDetailResponse struct {
+	Success  bool        `json:"success"`
+	Errors   []string    `json:"errors"`
+	Messages []string    `json:"messages"`
+	Result   WaitingRoom `json:"result"`
+}
+
+// WaitingRoomsResponse is the API response, containing an array of WaitingRooms.
+type WaitingRoomsResponse struct {
+	Success  bool          `json:"success"`
+	Errors   []string      `json:"errors"`
+	Messages []string      `json:"messages"`
+	Result   []WaitingRoom `json:"result"`
+}
+
+// CreateWaitingRoom creates a new Waiting Room for a zone.
+//
+// API reference: https://api.cloudflare.com/#waiting-room-create-waiting-room
+func (api *API) CreateWaitingRoom(ctx context.Context, zoneID string, waitingRoom WaitingRoom) (*WaitingRoom, error) {
+	uri := "/zones/" + zoneID + "/waiting_rooms"
```

Comment:
#626 consolidated URLs to use `fmt.Sprintf` so we'll update that here too

```suggestion
	uri := fmt.Sprintf("/zones/%s/waiting_rooms", zoneID)
```

Reply:
Fixed